File: update_subscription.py
#!/usr/bin/env python3
"""Update user subscription in Firestore - for testing only"""

import firebase_admin
from firebase_admin import credentials, firestore
from datetime import datetime, timedelta

# Initialize Firebase Admin SDK
cred = credentials.Certificate('serviceAccountKey.json')
firebase_admin.initialize_app(cred)

db = firestore.client()

# Find user by email
print('🔧 Updating subscription for alromn@yahoo.com...')

# Query users collection
users_ref = db.collection('users')
query = users_ref.where('email', '==', 'alromn@yahoo.com').limit(1)
results = query.get()

if not results:
    print('❌ User not found with email: alromn@yahoo.com')
    print('💡 Make sure the user has signed up in the app first')
    exit(1)

user_doc = results[0]
user_id = user_doc.id
user_data = user_doc.to_dict()
print(f"✅ Found user: {user_data.get('displayName', 'Unknown')} (ID: {user_id})")

# Update subscription to Pro tier
now = datetime.now()
trial_end = now + timedelta(days=14)
next_billing = now + timedelta(days=30)

subscription_data = {
    'subscription': {
        'tier': 'pro',
        'status': 'active',
        'startDate': firestore.SERVER_TIMESTAMP,
        'trialEndDate': trial_end,
        'nextBillingDate': next_billing,
        'billingCycle': 'monthly',
        'stripeCustomerId': f'test_customer_{user_id}',
        'stripeSubscriptionId': f'test_sub_{user_id}',
        'cancelAtPeriodEnd': False,
        'updatedAt': firestore.SERVER_TIMESTAMP,
    }
}

users_ref.document(user_id).update(subscription_data)

print('✅ Successfully updated subscription!')
print('📊 New subscription details:')
print('   Tier: Pro')
print('   Status: Active')
print(f'   Trial End: {trial_end.strftime("%Y-%m-%d %H:%M:%S")}')
print(f'   Next Billing: {next_billing.strftime("%Y-%m-%d %H:%M:%S")}')
print('🎉 alromn@yahoo.com now has Pro access!')
print('💡 Restart the app to see the changes')

File: update_subscription.dart
import 'package:firebase_core/firebase_core.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'firebase_options.dart';

/// Quick script to update subscription tier for a user
/// Run: dart run update_subscription.dart
void main() async {
  print('🔧 Updating subscription for alromn@yahoo.com...');

  // Initialize Firebase
  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);

  final firestore = FirebaseFirestore.instance;

  // Find user by email
  final usersQuery = await firestore
      .collection('users')
      .where('email', isEqualTo: 'alromn@yahoo.com')
      .limit(1)
      .get();

  if (usersQuery.docs.isEmpty) {
    print('❌ User not found with email: alromn@yahoo.com');
    print('💡 Make sure the user has signed up in the app first');
    return;
  }

  final userDoc = usersQuery.docs.first;
  final userId = userDoc.id;

  print('✅ Found user: ${userDoc.data()['displayName']} (ID: $userId)');

  // Update subscription to Pro tier
  final now = DateTime.now();
  final trialEnd = now.add(const Duration(days: 14));
  final nextBilling = now.add(const Duration(days: 30));

  await firestore.collection('users').doc(userId).update({
    'subscription': {
      'tier': 'pro',
      'status': 'active',
      'startDate': Timestamp.fromDate(now),
      'trialEndDate': Timestamp.fromDate(trialEnd),
      'nextBillingDate': Timestamp.fromDate(nextBilling),
      'billingCycle': 'monthly',
      'stripeCustomerId': 'test_customer_$userId',
      'stripeSubscriptionId': 'test_sub_$userId',
      'cancelAtPeriodEnd': false,
      'updatedAt': Timestamp.now(),
    },
  });

  print('✅ Successfully updated subscription!');
  print('📊 New subscription details:');
  print('   Tier: Pro');
  print('   Status: Active');
  print('   Trial End: ${trialEnd.toLocal()}');
  print('   Next Billing: ${nextBilling.toLocal()}');
  print('');
  print('🎉 alromn@yahoo.com now has Pro access!');
  print('💡 Restart the app to see the changes');
}
